src/Kis/Kis.hs

```diff
@@ -2,39 +2,69 @@
 {-# LANGUAGE OverloadedStrings          #-}
 {-# LANGUAGE RankNTypes                 #-}
 {-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeSynonymInstances       #-}
+{-# LANGUAGE FlexibleInstances          #-}
 
 module Kis.Kis
     ( withKis
-    , KisAction(..)
+    , KisRequest(..)
     , KisClient
     , KisConfig(..)
     , Kis(..)
     , KisException(..)
+    , KisT
+    , createBed
+    , createPatient
+    , getPatients
+    , getPatient
+    , placePatient
+    , runKis
     )
 where
 
 import Control.Monad.Logger
+import Control.Monad.Operational
 import Control.Monad.RWS hiding (asks)
 import Control.Monad.Reader
 import Database.Persist
-import Data.Text
+import Data.Text hiding (singleton)
 import GHC.Exception
 
 import Kis.Model
 import Kis.Time
 
-data KisAction a where
-    CreateBed :: String -> KisAction BedId
-    CreatePatient :: Patient -> KisAction PatientId
-    GetPatient :: PatientId -> KisAction (Maybe Patient)
-    GetPatients :: KisAction [Entity Patient]
-    PlacePatient :: PatientId -> BedId -> KisAction (Maybe PatientBedId)
+data KisRequest a where
+    CreateBed :: String -> KisRequest BedId
+    CreatePatient :: Patient -> KisRequest PatientId
+    GetPatient :: PatientId -> KisRequest (Maybe Patient)
+    GetPatients :: KisRequest [Entity Patient]
+    PlacePatient :: PatientId -> BedId -> KisRequest (Maybe PatientBedId)
 
-deriving instance Show (KisAction a)
+type KisT m = ProgramT KisRequest m
+
+runKis :: KisT IO a -> IO a
+runKis = undefined
+
+class Monad m => MonadKis m where
+    createBed :: String -> m BedId
+    createPatient :: Patient -> m PatientId
+    getPatients :: m [Entity Patient]
+    getPatient :: PatientId -> m (Maybe Patient)
+    placePatient :: PatientId -> BedId -> m (Maybe PatientBedId)
+
+instance Monad m => MonadKis (KisT m) where
+    createPatient = singleton . CreatePatient
+    createBed = singleton . CreateBed
+    getPatients = singleton GetPatients
+    getPatient = singleton . GetPatient
+    placePatient patId bedId = singleton (PlacePatient patId bedId)
+
+
+deriving instance Show (KisRequest a)
 
```

Comment:
Also ich habe gar nicht so viel verändert, denke ich. Vielleicht habe ich nur ungünstig Namen geändert.

Links die HEAD-Bezeichner, rechts der entsprechende Ausdruck in diesem PR.
```haskell
KisRequest a = KisRequest a
KisClient m a = KisT m a
req = singleton

 -- Nur für den "alten" `Kis` typen mit (KisRequest a -> m a) (ohne Clock)
runReaderT action kis = interpretWithMonadT kis action
```

Neu dazu gekommen sind `MonadKis`, die man in HEAD genauso definieren könnte und dann die folgende Instanz formulieren könnte
```haskell
instance Monad m => MonadKis (KisClient m) where
     createPatient = req . CreatePatient
     createBed = req . CreateBed
     getPatients = req GetPatients
     getPatient = req . GetPatient
     placePatient patId bedId = req (PlacePatient patId bedId)
```

Die `Clock` habe ich noch gar nicht untergebracht und Vorteile dieses Ansatzes sind auch noch fraglich. Ich finde den Typen `KisT m` besser als `KisClient m = ReaderT (Kis m) m`, weil er nur so etwas wie ein abstract syntax tree von den Konstruktoren aus `KisRequest a` ist und die Interpretation noch völlig offen lässt, während `ReaderT (Kis m) m` festlegt, dass wir eine Funktion `KisRequest a -> m a` benötigen. In der Implementierung hier habe ich die Forderung nach einer Funktion `KisRequest a -> m a` letztendlich immer noch drin, aber theoretisch sind andere Varianten möglich. Solange sich aber kein richtiger Nutzen ergibt, sollten wir es wohl besser lassen wie es ist. `operational` ist ja auch eine weitere Dependency..